### PR TITLE
RELATED: RAIL-3979 Make ErrorComponent and LoadingComponent mandatory

### PR DESCRIPTION
**Pull request by plavemar**
- DashboardComponentsContext ErrorComponent and LoadingComponent props made mandatory
- update components with IDashboardInsightProps

JIRA: RAIL-3979

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6458/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8307/ : FAILURE in 8m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6459/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8308/ : SUCCESS in 18m 11s

File: libs/sdk-ui-dashboard/.dependency-cruiser.js [hunk 1]
```diff
@@ -113,6 +113,7 @@ options = {
             "src/model/events/widget.ts",
             "src/presentation/constants",
             "src/presentation/dashboardContexts",
+            "src/presentation/dashboardContexts/DashboardComponentsContext.tsx",
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/.dependency-cruiser.js:116]: This should not be necessary, you are probably importing DashboardComponentsContext.tsx directly somewhere, not using the barell in dashboardContexts.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/InsightDrillDialog/InsightDrillDialog.tsx [hunk 2]
```diff
@@ -45,6 +48,15 @@ export const InsightDrillDialog = (props: InsightDrillDialogProps): JSX.Element
 
     const executionsHandler = useWidgetExecutionsHandler(DRILL_MODAL_EXECUTION_PSEUDO_REF);
 
+    const { ErrorComponent, LoadingComponent } = useDashboardComponentsContext({
+        ErrorComponent: ThrowMissingComponentError("ErrorComponent"),
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/InsightDrillDialog/InsightDrillDialog.tsx:52]: Do we need to "reset" the ErrorComponent as well? Maybe we should just not mention the ErrorComponent here at all and just override the Loading one? If so, then ThrowMissingComponentError can go back to not being exported.

File: libs/sdk-ui-dashboard/.dependency-cruiser.js (hunk 1 above)
Comment (plavemar) [reply libs/sdk-ui-dashboard/.dependency-cruiser.js]: Removed as the ThrowMissingComponentError does not need to be exported.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/InsightDrillDialog/InsightDrillDialog.tsx (hunk 2 above)
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/InsightDrillDialog/InsightDrillDialog.tsx]: Removed component overriding, it is not used in the child component and there is no need to do this.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6461/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8312/ : SUCCESS in 16m 27s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2149

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1895/ : SUCCESS in 58s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2176/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/979/ : SUCCESS in 17m 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8314/ : SUCCESS in 17m 01s

**Pull request merged**